django/views/decorators/debug.py

```diff
@@ -38,42 +38,43 @@ def my_function()
         )
 
     def decorator(func):
-        @wraps(func)
-        def sensitive_variables_wrapper(*func_args, **func_kwargs):
-            if variables:
-                sensitive_variables_wrapper.sensitive_variables = variables
-            else:
-                sensitive_variables_wrapper.sensitive_variables = "__ALL__"
-            return func(*func_args, **func_kwargs)
-
         if iscoroutinefunction(func):
-            markcoroutinefunction(sensitive_variables_wrapper)
+
+            @wraps(func)
+            async def sensitive_variables_wrapper(*func_args, **func_kwargs):
+                return await func(*func_args, **func_kwargs)
 
             wrapped_func = func
             while getattr(wrapped_func, "__wrapped__", None) is not None:
                 wrapped_func = wrapped_func.__wrapped__
 
             try:
                 file_path = inspect.getfile(wrapped_func)
-            except TypeError:  # raised for builtins
-                file_path = None
-
-            try:
-                _file_lines, first_file_line = inspect.getsourcelines(wrapped_func)
-            except TypeError:  # raised for builtins
-                first_file_line = None
-            assert file_path is not None and first_file_line is not None, (
-                f"{func.__name__} cannot safely be wrapped by "
-                + "@sensitive_variables_wrapper, please make it "
-                + "either non-async or defined in a file (and not a builtin)"
-            )
-            key = hash(f"{file_path}:{first_file_line}")
+                _, first_file_line = inspect.getsourcelines(wrapped_func)
+            except TypeError:  # Raises for builtins.
```

Comment:
Per your comment on the previous version of this code, maybe we should clarify this slightly:

```
... # Raises for builtins or native functions
```